pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/PulsarSource.java

```diff
@@ -142,8 +140,18 @@ public void open(Map<String, Object> config, SourceContext sourceContext) throws
                 .message(message)
                 .topicName(topicName)
                 .ackFunction(() -> {
-                    if (pulsarSourceConfig.getProcessingGuarantees() == FunctionConfig.ProcessingGuarantees.EFFECTIVELY_ONCE) {
-                        inputConsumer.acknowledgeCumulativeAsync(message);
+                    if (pulsarSourceConfig
+                            .getProcessingGuarantees() == FunctionConfig.ProcessingGuarantees.EFFECTIVELY_ONCE) {
+                        // try to find actual consumer of the messageId
+                        if (inputConsumer instanceof MultiTopicsConsumerImpl) {
```

Comment:
does this mean ackCumulativeAsync  is not working in MultiTopicsConsumerImpl? If so, should this be fixed in `MultiTopicsConsumerImpl` instead?

Reply:
>  If so, should this be fixed in MultiTopicsConsumerImpl instead?

Yeah, earlier I was thinking about ordering but MultiConsumer anyway doesn't maintain ordering among the consumers and we can also do cumulative ack for individual consumer. let me fix it.